### PR TITLE
Generate alert docs for website

### PR DESCRIPTION
**Pull request by psiinon**
I've tested the scripting in isolation but not all together in the workflow :/

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

File: .github/workflows/alert-docs.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        git update-index -q --refresh
+        # Generate a PR
+        cd zaproxy-website
+        git remote set-url origin https://$GITHUB_USER:$GITHUB_TOKEN@github.com/$GITHUB_USER/zaproxy-website.git
+        git add site/content/docs/alerts
+        git commit -m "Update alert pages" --signoff
+        git push --set-upstream origin master
+        hub pull-request --no-edit
+      env:
+        ZAPBOT_TOKEN: ${{ secrets.ZAPBOT_TOKEN }}
```
Comment (kingthorin) [inline .github/workflows/alert-docs.yml:39]: I know a lot of examples online put env after run, but doesn't it seem more logical to have this before the run block that uses it?

File: .github/workflows/alert-docs.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        export GITHUB_USER=zapbot
+        git config --global user.email "12745184+zapbot@users.noreply.github.com"
+        git config --global user.name $GITHUB_USER
+        # Run the ZAP script
+        docker run -v $(pwd):/zap/wrk/:rw --user root -t owasp/zap2docker-live ./zap.sh -addoninstallall -silent -script /zap/wrk/zap-admin/scripts/generate_alert_pages.js -cmd
+        # Update the index to be sure git is aware of changes
+        git update-index -q --refresh
+        # Generate a PR
+        cd zaproxy-website
+        git remote set-url origin https://$GITHUB_USER:$GITHUB_TOKEN@github.com/$GITHUB_USER/zaproxy-website.git
```
Comment (kingthorin) [inline .github/workflows/alert-docs.yml:33]:
```suggestion
        git remote set-url origin https://$GITHUB_USER:$ZAPBOT_TOKEN@github.com/$GITHUB_USER/zaproxy-website.git
```

File: .github/workflows/alert-docs.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        git add site/content/docs/alerts
+        git commit -m "Update alert pages" --signoff
+        git push --set-upstream origin master
```
Comment (thc202) [inline .github/workflows/alert-docs.yml:36]: Better push to other branch than master (e.g. master:update-alerts).

File: .github/workflows/alert-docs.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  update-alerts:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        path: zap-admin
+        fetch-depth: 10
```
Comment (thc202) [inline .github/workflows/alert-docs.yml:13]: Can be removed for this case.

File: .github/workflows/alert-docs.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    - name: Update alert docs
+      run: |
+        # Setup git details
+        export GITHUB_USER=zapbot
+        git config --global user.email "12745184+zapbot@users.noreply.github.com"
+        git config --global user.name $GITHUB_USER
+        # Run the ZAP script
+        docker run -v $(pwd):/zap/wrk/:rw --user root -t owasp/zap2docker-live ./zap.sh -addoninstallall -silent -script /zap/wrk/zap-admin/scripts/generate_alert_pages.js -cmd
+        # Update the index to be sure git is aware of changes
+        git update-index -q --refresh
```
Comment (thc202) [inline .github/workflows/alert-docs.yml:30]: Should be done after the cd.

**Comment by psiinon** [comment]
Hopefully fixed all of those

**Comment by thc202** [comment]
Thank you!

**Pull request merged**